### PR TITLE
#225 Add type for feature update note

### PR DESCRIPTION
**Pull request by LanHuong142002**
- Add type for feature update note
- Remove redundant condition
- User `!` to avoid null case when query DOM

This PR will block PR #222 
This PR is blocked by PR #230

### CODE REVIEW

File: typescript/big-practice/src/scripts/controllers/noteController.ts [hunk 1]
```diff
@@ -92,28 +92,34 @@ export default class NoteController {
...
 
       // function delete note in tab trash
       this.view.bindDeleteNoteInTrash(async (id) => {
-        this.loadingPage.addLoading();
-        await this.model.deleteNoteInTrash(id);
-
-        this.view.removeNoteElement(id);
-        this.view.showHideEmpty(this.model.listNotes, NOTE.TRASH_NOTES);
-        this.loadingPage.removeLoading();
+        if (id) {
```
Comment (phuongbuimai-agilityio) [inline typescript/big-practice/src/scripts/controllers/noteController.ts]: Why do we need check this condition?

File: typescript/big-practice/src/scripts/views/listNoteView.ts [hunk 2]
```diff
@@ -221,7 +221,7 @@ export default class ListNoteView {
   removeNoteElement(id: string): void {
     const note = selectDOMById(id);
 
-    if (note) {
+    if (note && id) {
```
Comment (phuongbuimai-agilityio) [inline typescript/big-practice/src/scripts/views/listNoteView.ts]: Why do we need to check the id in this condition?

File: typescript/big-practice/src/scripts/views/listNoteView.ts [hunk 3]
```diff
@@ -231,21 +231,25 @@ export default class ListNoteView {
...
-        descriptionElement.textContent = '';
-      } else {
-        this.elementHelpers.addClass(emptyNoteElement, 'hide');
-        titleElement.textContent = noteItem.title;
-        descriptionElement.textContent = noteItem.description;
+      const emptyNoteElement = note.querySelector(
+        '.note-content .note-empty'
+      ) as HTMLElement;
+
+      if (titleElement && descriptionElement) {
```
Comment (phuongbuimai-agilityio) [inline typescript/big-practice/src/scripts/views/listNoteView.ts]: Please help explain this condition

File: typescript/big-practice/src/scripts/views/listNoteView.ts [hunk 4]
```diff
@@ -81,26 +78,22 @@ export default class ListNoteView {
       message: 'Notes you add appear here',
     };
 
-    if (this.sectionWrapper) {
-      if (sessionStorage.getItem(STORAGE_KEYS.PAGE_NUMBER) === '0') {
-        this.sectionWrapper.innerHTML = '';
-        this.sectionWrapper.appendChild(formAddNote());
+    if (sessionStorage.getItem(STORAGE_KEYS.PAGE_NUMBER) === '0') {
```
Comment (thanhnguyen-agilityio) [inline typescript/big-practice/src/scripts/views/listNoteView.ts:81]: Please explain me this `sessionStorage.getItem(STORAGE_KEYS.PAGE_NUMBER) === '0'`

**Pull request merged**